### PR TITLE
Add constantLine function

### PR DESCRIPTION
**Pull request by agao48**
**Describe your changes**
Adds the constantLine function

**Testing performed**
Test files added with [mtcmptest](https://github.com/shanson7/mtcmptest) used locally

### CODE REVIEW

**Comment by replay** [comment]
There are some formatting issues which make the circleCI tests fail

File: expr/func_constantline.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	return context
+}
+
+func (s *FuncConstantLine) Exec(dataMap DataMap) ([]models.Series, error) {
+	out := pointSlicePool.Get().([]schema.Point)
+	out = append(out,
+		schema.Point{Val: s.value, Ts: s.from},
+		schema.Point{Val: s.value, Ts: s.from + uint32((s.to-s.from)/2.0)},
+		schema.Point{Val: s.value, Ts: s.to},
+	)
```
Comment (Dieterbe) [inline expr/func_constantline.go]: it is guaranteed at this point (due to prior validation) that from < to.
however, from may equal to -1, which would result in duplicate timestamps. this is an edge case, and causes graphite to crash, but we may as well handle it :)

**Comment by agao48** [comment]
I added https://github.com/bloomberg/metrictank/commit/38e1c21ca69d4acf1827733b260490ffb674012c 10 minutes ago, but not showing up here. There was some wonky stuff that happened over weekend where the original branch got deleted, so if the commit doesn't show up here by EOD, I'll close this PR and make a new one.

**Comment by agao48** [comment]
closing per comment above ^ - reopening

**Pull request closed without merging**